kroxylicious-filter/src/main/java/io/strimzi/kafka/topicenc/kroxylicious/FetchDecryptFilter.java

```diff
@@ -0,0 +1,128 @@
+     */
+    private void resolveTopicsAndReturnError(ResponseHeaderData header, FetchResponseData response, KrpcFilterContext context) {
+        Stream<Uuid> topicIdsToResolve = response.responses().stream().filter(Predicate.not(this::isResolvable)).map(FetchResponseData.FetchableTopicResponse::topicId);
+        // fire-and-forget a metadata request to resolve topic ids, preparing for future fetches
+        resolveAndCache(context, topicIdsToResolve);
+        FetchResponseData data = new FetchResponseData();
+        data.setErrorCode(Errors.UNKNOWN_SERVER_ERROR.code());
+        context.forwardResponse(header, data);
+    }
+
+    private void resolveAndCache(KrpcFilterContext context, Stream<Uuid> topicIdsToResolve) {
+        MetadataRequestData request = new MetadataRequestData();
+        topicIdsToResolve.forEach(uuid -> {
+            MetadataRequestData.MetadataRequestTopic e = new MetadataRequestData.MetadataRequestTopic();
+            e.setTopicId(uuid);
+            request.topics().add(e);
+        });
+        // if the client is sending topic ids we will assume the broker can support at least the lowest metadata apiVersion
+        // supporting topicIds
+        CompletionStage<MetadataResponseData> stage = context.sendRequest(METADATA_VERSION_SUPPORTING_TOPIC_IDS, request);
+        stage.thenAccept(response -> cacheTopicIdToName(response, METADATA_VERSION_SUPPORTING_TOPIC_IDS));
+    }
+
+    private void decryptFetchResponse(ResponseHeaderData header, FetchResponseData response, KrpcFilterContext context) {
+        for (FetchResponseData.FetchableTopicResponse fetchResponse : response.responses()) {
+            Uuid originalUuid = fetchResponse.topicId();
+            String originalName = fetchResponse.topic();
+            if (originalName == null || originalName.equals("")) {
+                fetchResponse.setTopic(topicUuidToName.get(originalUuid));
+                fetchResponse.setTopicId(null);
+            }
+            try {
+                module.decrypt(fetchResponse);
+            } catch (Exception e) {
+                log.error("Failed to decrypt a fetchResponse for topic: " + fetchResponse.topic(), e);
+                throw new RuntimeException(e);
+            }
+            fetchResponse.setTopic(originalName);
+            fetchResponse.setTopicId(originalUuid);
+        }
+        context.forwardResponse(header, response);
+    }
+
+
+    private boolean isResolvable(FetchResponseData.FetchableTopicResponse fetchableTopicResponse) {
+        return (fetchableTopicResponse.topic() != null && !fetchableTopicResponse.topic().equals("")) || topicUuidToName.containsKey(fetchableTopicResponse.topicId());
+    }
+
+    private boolean isResolvable(FetchRequestData.FetchTopic fetchTopic) {
+        return (fetchTopic.topic() != null && !fetchTopic.topic().equals("")) || topicUuidToName.containsKey(fetchTopic.topicId());
```

Comment:
```suggestion
        return (fetchTopic.topic() != null && !fetchTopic.topic().strip().equals("")) || topicUuidToName.containsKey(fetchTopic.topicId());
```

Given its the third time I've made this suggestion maybe we should extract a function to check the name is useful?

Reply:
done, I've made a `Strings` class with `isNullOrBlank(String originalName)` doing the strip.